Sming/Libraries/SDCard/SDCard.cpp

```diff
@@ -33,27 +33,39 @@ Descr: Low-level SDCard functions
 
 /-------------------------------------------------------------------------*/
 #include "SDCard.h"
-#include "fatfs/diskio.h"	/* Declarations of disk I/O functions */
-
-FATFS *pFatFs = NULL;		/* FatFs work area needed for each volume */
-SPIBase *SDCardSPI = NULL;
-uint8 SPI_CS;				/* SPI client selector */
+#include "fatfs/diskio.h" /* Declarations of disk I/O functions */
 
+FATFS* pFatFs = NULL; /* FatFs work area needed for each volume */
+SPIBase* SDCardSPI = NULL;
+uint8 SPI_CS;		  /* SPI client selector */
+uint8 SPI_byteOrder;  /* The order the SPI bytes are sent */
+uint32 SPI_freqLimit; /* The max frequency the SPI should run */
+uint32 SPI_init_freq; /* SPI frequency usined for init */
```

Comment:
Private/internal variables should either be `static` or, preferably, be defined within an un-named namespace.
The code might be clearer if these were within a structure:

```
namespace {

struct SpiConfig{
  uint8_t cs;
  uint8_t byteOrder;
  uint8_t freqLimit;
  uint8_t initFreq;
};

SpiConfig spi;

};
```


Variables names should be camel-case, especially `SPI_CS` which looks like a #defined constant.
Also, in C++ `nullptr` is preferred to `NULL`.